Document behavior of `prefix` in dependabot.yml (#22884)
github/docs: content/code-security/dependabot/dependabot-version-updates/configuration-options-for-the-dependabot.yml-file.md

{% endnote %}

- `prefix` specifies a prefix for all commit messages.

>    When you specify a prefix for commit messages, {% data variables.product.prodname_dotcom %} will automatically add a colon between the defined prefix and the commit message, unless you end the end the prefix with a space.

Reviewer: @schra - I have talked with the engineering team and we agreed that we didn't want to document the regexp in the docs. First of all, it doesn't fit in with our content model for the docs, and also it will be a maintenance overhead if/when we may change the implementation. I am therefore suggesting the following:

```suggestion
   When you specify a prefix for commit messages, {% data variables.product.prodname_dotcom %} will automatically add a colon between the defined prefix and the commit message provided the defined prefix ends with a letter, number, closing parenthesis, or closing bracket. This means that, for example, if you end the prefix with a whitespace, there will be no colon added between the prefix and the commit message.
```

Author: Sounds good to me!

   The code snippet below provides examples of both in the same configuration file.

> 

Author: > Can you add the following to that line, after prefix specifies a prefix for all commit messages.?
>
>When you specify a prefix for commit messages, {% data variables.product.prodname_dotcom %} will automatically add a colon between the defined prefix and the commit message, unless you end the end the prefix with a space. The code snippet below provides examples of both in the same configuration file.

I added this now, but I think this is false. Let's look at the implementation (https://github.com/dependabot/dependabot-core/blob/06702c83e5d964173504ddba6838c37fe37207ba/common/lib/dependabot/pull_request_creator/pr_name_prefixer.rb#L75-L83):

```
prefix += ":" if prefix.match?(/[A-Za-z0-9\)\]]\Z/)
```

So having whitespace at the end is one way of many ways to "disable" the colon. For example, if your prefix contains an umlaut (eg ö), then there also is no colon. 

Do you have an idea how to formulate this, so that this is more clear? Should we maybe add the regex to the documentation, so it's very clear when a colon is added?

    schedule:
      interval: "weekly"
    commit-message:

>       # Prefix all commit messages with "npm: "

Reviewer: Should this be like in the suggestion?
```suggestion
      # Prefix all commit messages with "npm:"
```

Reviewer: Argh, I got this wrong. Let me marked this as resolved 🤦🏻

      prefix: "npm"

  - package-ecosystem: "docker"
    directory: "/"
    schedule:
      interval: "weekly"
    commit-message:

>       # Prefix all commit messages with "[docker] " (no colon)

Reviewer: ```suggestion
      # Prefix all commit messages with "[docker] " (no colon, but a trailing whitespace)
```

    schedule:
      interval: "weekly"
    commit-message:
      # Prefix all commit messages with "npm: "
      prefix: "npm"

  - package-ecosystem: "docker"
    directory: "/"
    schedule:
      interval: "weekly"
    commit-message:
      # Prefix all commit messages with "[docker] " (no colon, but a trailing whitespace)
      prefix: "[docker] "

  - package-ecosystem: "composer"
    directory: "/"
    schedule:
      interval: "weekly"

>     # Prefix all commit messages with "Composer" plus its scope, i. e. a list

Reviewer: We don't use `i.e.` in the GitHub docs 
```suggestion
    # Prefix all commit messages with "Composer" plus its scope, such as a list
```

Author: Oh okay, I wasn't aware of that.

I'm not a native speaker, but imho "such as" sounds wrong here. I think it should be "that is", because I don't want to give an example, but I want to define what a "scope" is. I updated `such as` to `that is` now. If you have concerns about that, feel free to revert my commit :)

Reviewer: I am happy with your commit 😃 👍🏻
